daml-foundations/daml-ghc/ide/src/Development/IDE/Core/RuleTypes/Daml.hs

```diff
@@ -35,7 +35,16 @@ type instance RuleResult GenerateRawDalf = LF.Module
 type instance RuleResult GeneratePackage = LF.Package
 type instance RuleResult GenerateRawPackage = LF.Package
 type instance RuleResult GeneratePackageDeps = LF.Package
-type instance RuleResult GeneratePackageMap = (Map UnitId (LF.PackageId, LF.Package, BS.ByteString, FilePath))
+
+data DalfPackage = DalfPackage
+    { dalfPackageId :: LF.PackageId
+    , dalfPackagePkg :: LF.Package
+    , dalfPackageBytes :: BS.ByteString
```

Comment:
Do you know why we need both the deserialized package and the serialized bytes?

Reply:
We need the deserialized package for things like `initWorld`. I believe the `ByteString` is only used for building the DAR. We could reencode the `Package` here but that seems somewhat silly so for now I would keep it this way.